File: ParallelBatchEval/Py/TSP.py
import numpy as np
import pickle

ixcoord = []
iycoord = []

# read from input file
def read_input(instname):
    filename="../../instances/tsp/"+instname+".tsp"
    with open(filename, 'r') as f:
        content = f.read()
        lines = content.split('\n')
        # print(lines)

        dim = int(lines[3].split()[-1])

        for line in lines[6:]:
            if line.startswith('EOF'):
                break
            else:
                X, Y = line.split()[1:]
                ixcoord.append(float(X))
                iycoord.append(float(Y))

        xcoord = np.zeros(dim,dtype=float)
        ycoord = np.zeros(dim,dtype=float)

        for i in range(0,dim):
            xcoord[i] = ixcoord[i]
            ycoord[i] = iycoord[i]

    return xcoord,ycoord,dim

File: ParallelBatchEval/Julia/TSP.jl
module TSP
    using DelimitedFiles

	dim = Int(0)

    xcoord = Array{Float64,1}
    ycoord = Array{Float64,1}

	export read_in_TSP,evaluate_TSP
	export dim,xcoord,ycoord

#####################################
	function read_in_TSP(instname)
		filename = string("../../instances/tsp/",instname,".tsp")
		# read from file...
		file = open(filename)

        title = readline(file, keep=false)
        tsp_type = readline(file, keep=false)
        comment = readline(file, keep=false)

        dimString = readline(file, keep=true)
        global dim = parse(Int64,split(dimString,":")[2])

        edge_weight = readline(file, keep=false)
        dummy = readline(file, keep=false)

        index = 1

        global xcoord = zeros(dim)
        global ycoord = zeros(dim)

        while index <= dim #eof(file) != true
            tmp = readline(file, keep=true)
            # println(tmp)
            global xcoord[index] = parse(Float64, split(tmp," ")[2])
            global ycoord[index] = parse(Float64, split(tmp," ")[3])
            index += 1
        end

        close(file)
    end
#####################################

    function dist(xcoord,ycoord,a,b)
        dx = xcoord[a]-xcoord[b]
        dy = ycoord[a]-ycoord[b]
        return dx*dx + dy*dy
    end

#####################################
	function evaluate_TSP(perm,xcoord,ycoord,dim)
        cost = sqrt(dist(xcoord,ycoord,perm[1],perm[2]))
        for i=2:dim-1
            cost += sqrt(dist(xcoord,ycoord,perm[i],perm[i+1]))
        end
        cost += sqrt(dist(xcoord,ycoord,perm[dim],perm[1]))
        return cost
	end
#####################################
end
